autoload/neomake.vim

```diff
@@ -142,6 +144,23 @@ function! s:MakeJob(make_id, options) abort
         let maker = call(maker.fn, [jobinfo], maker)
     endif
 
+    if !executable(maker.exe)
+        if !get(maker, 'auto_enabled', 0)
+            let error = printf('Exe (%s) of maker %s is not executable.', maker.exe, maker.name)
+            if !has_key(s:exe_error_thrown, maker.exe)
+                let s:exe_error_thrown[maker.exe] = 1
+                throw 'Neomake: '.error
+            endif
+            call neomake#utils#DebugMessage(error)
+            return -1
+        endif
+
+        " XXX: mark it as to be skipped earlier?!
```

Comment:
The common use case is that those would be skipped always, so we should short-circuit it.